Source/Urho3D/UI/ToolTip.cpp

```diff
@@ -74,7 +74,7 @@ void ToolTip::Update(float timeStep)
             SharedPtr<UIElement> target = it->Lock();
             if (!target)
                 it = altTargets_.Erase(it);
-            else if (hovering = target->IsHovering() && target->IsVisibleEffective())
+            else if ((hovering = target->IsHovering() && target->IsVisibleEffective()))
```

Comment:
I don't understand what the extra parentheses around are changing

Reply:
See [this reference](https://stackoverflow.com/questions/4369446/using-the-result-of-an-assignment-as-a-condition-without-parentheses)

Reply:
Thanks

Reply:
Both options are fine for me, but consider actually fixing `if` instead of shutting compiler warning with extra braces.
E.g. `else if (hovering = x) {` -> `else { hovering = x; if (hovering) {`
IMO it is bad practice to mix actions and conditions in single line for sake of shorter code.

Reply:
Good point @eugeneko, updated patch